akka-docs/rst/scala/code/docs/akka/typed/IntroSpec.scala

```diff
@@ -0,0 +1,206 @@
+        case AddAndGet(amount, replyTo) ⇒
+          val nextCount = count + amount
+          replyTo ! CurrentCount(nextCount)
+          behavior(nextCount)
+        case GetAndAdd(amount, replyTo) ⇒
+          replyTo ! CurrentCount(count)
+          behavior(count + amount)
+      }
+  }
+  //#counter-actor
+
+  //#chatroom-actor
+  object ChatRoom {
+    //#chatroom-protocol
+    sealed trait Command
+    final case class GetSession(screenName: String, replyTo: ActorRef[SessionEvent])
+      extends Command
+    //#chatroom-protocol
+    //#chatroom-behavior
+    private final case class PostSessionMessage(screenName: String, message: String)
+      extends Command
+    //#chatroom-behavior
+    //#chatroom-protocol
+
+    sealed trait SessionEvent
+    final case class SessionGranted(handle: ActorRef[PostMessage]) extends SessionEvent
+    final case class SessionDenied(reason: String) extends SessionEvent
+    final case class MessagePosted(screenName: String, message: String) extends SessionEvent
+
+    final case class PostMessage(message: String)
+    //#chatroom-protocol
+    //#chatroom-behavior
+
+    val behavior: Behavior[GetSession] =
+      ContextAware[Command] { ctx ⇒
+        var sessions = List.empty[ActorRef[SessionEvent]]
+
+        Static {
+          case GetSession(screenName, client) ⇒
+            sessions ::= client
+            val wrapper = ctx.spawnAdapter {
+              p: PostMessage ⇒ PostSessionMessage(screenName, p.message)
+            }
+            client ! SessionGranted(wrapper)
+          case PostSessionMessage(screenName, message) ⇒
+            val mp = MessagePosted(screenName, message)
+            sessions foreach (_ ! mp)
+        }
+      }.narrow // only expose GetSession to the outside
+    //#chatroom-behavior
+  }
+  //#chatroom-actor
+
+}
+
+class IntroSpec extends TypedSpec {
+  import IntroSpec._
+
+  def `must say hello`(): Unit = {
+    //#hello-world
+    import HelloWorld._
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    val system: ActorSystem[Greet] = ActorSystem("hello", Props(greeter))
+
+    val future: Future[Greeted] = system ? (Greet("world", _))
+
+    for {
+      greeting <- future
+      done <- { println(s"result: $greeting"); system.terminate() }
+    } println("system terminated")
+    //#hello-world
+  }
+
+  def `must count with partner`(): Unit = {
+    //#counter-become
+    import Counter._
+
+    // first expect a current count of 0
+    def first(c: ActorRef[Command]) = {
+      Full[CurrentCount] {
+        case Msg(ctx, current) ⇒
+          println(s"current count was ${current.x}")
+          c ! AddAndGet(6, ctx.self)
+          second(c)
+      }
+    }
+    // then expect a changed count of 11
+    def second(c: ActorRef[Command]) =
+      Total[CurrentCount] { current ⇒
+        println(s"now the count was ${current.x}")
+        Stopped
+      }
+
+    val system = ActorSystem("counterBecome", Props(Full[CurrentCount] {
```

Comment:
I forgot to delete this test case; the chat room proved to be a better example for the docs.